test/operations/candidates/upload_file_test.rb

```diff
@@ -0,0 +1,240 @@
+    )
+
+    Candidates::UploadFile.new(
+      candidate:,
+      file:,
+      source: "linkedin",
+      namespace: :api
+    ).call.value!
+
+    candidate.reload
+
+    assert_equal candidate.files.size, 1
+
+    file = candidate.files.first
+
+    assert_nil file.blob.custom_metadata[:text_checksum]
+    assert_nil file.blob.custom_metadata[:source]
+  end
+
+  test "should mark the file as cv when we upload a pdf file with cv flag and the candidate did not have cv " \
+       "or existing cv is from the same source" do
+    candidate = candidates(:sam)
+
+    assert_nil candidate.cv
+
+    Candidates::UploadFile.new(
+      candidate:,
+      file: @empty_pdf_file,
+      cv: true,
+      source: "linkedin",
+      namespace: :api
+    ).call.value!
+
+    candidate.reload
+
+    old_cv = candidate.cv
+
+    assert_predicate old_cv, :present?
+
+    Candidates::UploadFile.new(
+      candidate:,
+      file: @not_empty_pdf_file,
+      cv: true,
+      source: "linkedin",
+      namespace: :api
+    ).call.value!
+
+    candidate.reload
+
+    new_cv = candidate.cv
+
+    assert_predicate new_cv, :present?
+    assert_not_equal old_cv, new_cv
+    assert_equal candidate.files.size, 2
+  end
+
+  test "should not mark the file as cv when we upload a pdf file with cv flag via api " \
+       "and the candidate has a cv from another source" do
+    candidate = candidates(:sam)
+
+    assert_nil candidate.cv
+
+    Candidates::UploadFile.new(
+      candidate:,
+      file: @empty_pdf_file,
+      cv: true,
+      source: "reed",
+      namespace: :api
+    ).call.value!
+
+    candidate.reload
+
+    old_cv = candidate.cv
+
+    assert_predicate old_cv, :present?
+
+    Candidates::UploadFile.new(
+      candidate:,
+      file: @not_empty_pdf_file,
+      cv: true,
+      source: "linkedin",
+      namespace: :api
+    ).call.value!
+
+    candidate.reload
+
+    new_cv = candidate.cv
+
+    assert_predicate new_cv, :present?
+    assert_equal old_cv, new_cv
+    assert_equal candidate.files.size, 2
+
+    file1_metadata = candidate.files.first.blob.custom_metadata
+    file2_metadata = candidate.files.second.blob.custom_metadata
+
+    assert_not_equal file1_metadata[:source], file2_metadata[:source]
+    assert_not_equal file1_metadata[:text_checksum], file2_metadata[:text_checksum]
+  end
+
+  test "should mark the file as cv when we upload a pdf file with cv flag via ats " \
+       "and the candidate has a cv from another source" do
+    candidate = candidates(:sam)
+
+    assert_nil candidate.cv
+
+    Candidates::UploadFile.new(
+      candidate:,
+      file: @empty_pdf_file,
+      cv: true,
+      source: "reed",
+      namespace: :api
+    ).call.value!
+
+    candidate.reload
+
+    old_cv = candidate.cv
+
+    assert_predicate old_cv, :present?
+
+    Candidates::UploadFile.new(
+      candidate:,
+      file: @not_empty_pdf_file,
+      cv: true,
+      source: "linkedin",
```

Comment:
Could you please remind me how do we determine a source when we upload a file manually (via ats)?

Reply:
via interface we did not set or calculate the `source`
decided not to add this logic.

Reply:
I kinda don't follow. Why do you specify source on manual uploading? Shouldn't it be blank?

Reply:
updated